Resources/public/js/views/fields/ez-country-editview.js

```diff
@@ -0,0 +1,161 @@
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-country-editview', function (Y) {
+    "use strict";
+    /**
+     * Provides the field edit view for the country fields
+     *
+     * @module ez-country-editview
+     */
+
+    Y.namespace('eZ');
+
+    var FIELDTYPE_IDENTIFIER = 'ezcountry';
+
+    /**
+     * Country edit view
+     *
+     * @namespace eZ
+     * @class CountryEditView
+     * @constructor
+     * @extends eZ.SelectionEditView
+     */
+    Y.eZ.CountryEditView = Y.Base.create('countryEditView', Y.eZ.SelectionEditView, [], {
+
+        /**
+         * Returns the current country name.
+         * If the country is missing, it will return a country code and send a warning.
+         *
+         * @method _getCountryName
+         * @param {String} alpha2Code The country code (Alpha2)
+         * @protected
+         * @return {String}
+         */
+        _getCountryName: function (alpha2Code) {
+            var countryList = this.get('config');
+
+            if (countryList[alpha2Code]) {
+                return countryList[alpha2Code].Name;
+            } else {
+                console.warn("Unknown country code: " + alpha2Code);
+                return alpha2Code;
+            }
+        },
+
+        _getSelectionFilter: function () {
+            var container = this.get('container'),
+                selectedObjectArray = this._getSelectedTextValues(),
+                input = container.one('.ez-selection-filter-input'),
+                source = this._getSource(),
+                selected = [];
+
+            Y.Array.each(selectedObjectArray, function (selectedObject) {
+                selected.push(selectedObject.text);
+            });
+            return new Y.eZ.SelectionFilterView({
+                container: input.get('parentNode'),
+                inputNode: input,
+                listNode: this.get('container').one('.ez-selection-options'),
+                selected: selected,
+                source: source,
+                filter: true,
+                resultFilters: 'startsWith',
+                resultHighlighter: 'startsWith',
+                isMultiple: this.get('isMultiple'),
+                resultTextLocator: function (sourceElement) {
+                    return sourceElement.Name;
+                },
+                resultAttributesFormatter: function (sourceElement) {
+                    return {
+                        alpha2: sourceElement.Alpha2,
+                        text: sourceElement.Name,
+                    };
+                },
+            });
+        },
+
+        _addSelection: function (attributes) {
+            var values = this.get('values');
+
+            values.push({text:attributes.text , alpha2: attributes.alpha2});
+            this._set('values', values, {"added": attributes.text});
+        },
+
+        _removeSelection: function (text, node) {
```

Comment:
to be consistent with `_addSelection`, `_removeSelection` should probably take the attributes in parameter (and the search through `values` should be made against the alpha2 code as well).